src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509chain.c

```diff
@@ -0,0 +1,484 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_x509chain.h"
+#include <pal_x509_types.h>
+
+#include <assert.h>
+#include <stdbool.h>
+#include <string.h>
+
+#define INIT_LOCALS(name, ...) \
+    enum { __VA_ARGS__, count_##name }; \
+    jobject name[count_##name] = { 0 }; \
+
+#define RELEASE_LOCALS(name, env) \
+{ \
+    for (int i_##name = 0; i_##name < count_##name; ++i_##name) \
+    { \
+        jobject local = name[i_##name]; \
+        if (local != NULL) \
+            (*env)->DeleteLocalRef(env, local); \
+    } \
+} \
```

Comment:
At this point do we want to unify all of the definitions of these macros we have into one definition in pal_jni?